internal/database/database.go

```diff
@@ -204,6 +217,85 @@ func (d *cosmosDBClient) GetLockClient() *LockClient {
 	return d.lockClient
 }
 
+func (d *cosmosDBClient) getBillingID(ctx context.Context, resourceID *azcorearm.ResourceID) (string, error) {
+	pk := NewPartitionKey(resourceID.SubscriptionID)
+
+	// Resource ID alone does not uniquely identify a billing document, but
```

Comment:
What is the primary key?

Reply:
Primary key is the `ID` field.

Reply:
Correct.  Maybe it won't be an issue but I didn't want to assume the billing service will expunge a billing document for a deleted cluster before the RP might have to create a new one with the same resource ID.

Reply:
I _think_ I answered your question but feel free to unresolve if additional discussion is needed.